spec/controllers/users/mfa_selection_controller_spec.rb

```diff
@@ -0,0 +1,156 @@
+require 'rails_helper'
+
+describe Users::MfaSelectionController do
+  before do
+    allow(IdentityConfig.store).to receive(:select_multiple_mfa_options).and_return(true)
+    user = build(:user, :signed_up)
+    stub_sign_in(user)
+  end
+  let(:current_sp) { create(:service_provider) }
+
+  describe 'GET index' do
+    context 'when the user is using one authenticator option' do
+      it 'shows the mfa setup screen' do
+        controller.user_session[:selected_mfa_options] = ['backup_code']
+
+        get :index
+
+        expect(response).to render_template(:index)
+      end
+    end
+  end
+
+  describe 'Submitting the form' do
```

Comment:
style, I'd name this after the method
```suggestion
  describe '#update' do
```